File: spec/cypress/integration/new_messenger/event_triggering_spec.js
import {
  translations
} from '../../support'

describe('Customization Spec', function () {
  beforeEach(() => {
    cy.appEval('ActiveJob::Base.queue_adapter = :test')
    cy.appEval('ActiveJob::Base.queue_adapter.perform_enqueued_at_jobs = true')
  })

  it.only('wakeup & toggle method', function () {
    cy.appScenario('basic')

    translations()

    cy.appEval('App.last').then((results) => {
      const appKey = results.key

      cy.visit(`/tester/${appKey}`)
        .then(() => {
          cy.get('iframe:first')
            .then(function ($iframe) {
              const $body = $iframe.contents().find('body')
              // cy.wrap($body).find("#chaskiq-prime").click()
            })

          cy.window().then((win) => {
            cy.wait(2000)
            cy.get('iframe:first')
              .then(function ($iframe) {
                const $body = $iframe.contents().find('body')
                //console.log($body.html())
                //expect($body.html()).to.not.contain('Start a conversation')
                cy.get("#frame-wrapper[data-open=false]") //.should('not.be.visible')
              }).then(() => {
            
                 win.chaskiqMessenger.sendCommand('wakeup', {})

                 cy.get("#frame-wrapper[data-open=true]").then(()=>{

                  win.chaskiqMessenger.sendCommand('toggle', {})
                  cy.get("#frame-wrapper[data-open=false]")

                 })
   
              })
          })
        })
    })
  })
})

File: spec/system/messenger/event_triggering_spec.rb
require "rails_helper"

RSpec.describe "Event Triggering", type: :system, js: true do
  let!(:app) do
    Plugin.restore_plugins_from_fs

    FactoryBot.create(:app, encryption_key: "unodostrescuatro",
                            active_messenger: "true",
                            state: "enabled",
                            domain_url: "*")
  end

  before do
    # Setup for ActiveJob and any other initial conditions
    ActiveJob::Base.queue_adapter = :test
    ActiveJob::Base.queue_adapter.perform_enqueued_at_jobs = true
  end

  it "wakeup & toggle method" do
    visit "/tester/#{app.key}"

    # open_new_messenger({params: { lang: 'en' }}) do |page|
    # The following steps simulate the Cypress steps within the iframe context
    # Note that direct execution of custom JS commands may vary in Capybara

    sleep(2)
    # Execute 'wakeup' command
    page.execute_script("window.chaskiqMessenger.sendCommand('wakeup', {})")
    sleep(1) # Wait for any changes to take effect

    # Assert the frame-wrapper data attribute to be true
    expect(page).to have_css("#frame-wrapper[data-open='true']")

    # Execute 'toggle' command
    page.execute_script("window.chaskiqMessenger.sendCommand('toggle', {})")
    sleep(1) # Wait for any changes to take effect

    # Assert the frame-wrapper data attribute to be false

    # expect(page).to have_css("#frame-wrapper[data-open='false']")
    expect(page.body.include?('<div id="frame-wrapper" data-open="false" ')).to be == true
    # end
  end
end
